Description: Fix build failure with newer glibc
 Currently the package FTBFS because of errors re: the deprecated __malloc_hook
 etc. usage. The fix is basically taken from 'upstream' - I mention 'upstream'
 as right now it's not really clear which repository can be considered upstream
 for this project.
Origin: upstream, https://github.com/gordan-bobic/zfs-fuse/commit/9d554eb82861d449f5e86aa6c272da6dd91377a8

--- a/src/lib/libumem/malloc.c
+++ b/src/lib/libumem/malloc.c
@@ -441,29 +441,9 @@
 	return (buf);
 }
 
-#ifdef __GLIBC__
-static void __attribute__((constructor)) umem_malloc_init_hook(void)
-{
-	if (__malloc_hook != umem_malloc_hook) {
-		umem_startup(NULL, 0, 0, NULL, NULL);
-		__malloc_hook = umem_malloc_hook;
-		__free_hook = umem_free_hook;
-		__realloc_hook = umem_realloc_hook;
-		__memalign_hook = umem_memalign_hook;
-	}
-}
-
-#if __GLIBC_PREREQ(2, 14)
-void (* __volatile __malloc_initialize_hook)(void) = umem_malloc_init_hook;
-#else
-void (* __malloc_initialize_hook)(void) = umem_malloc_init_hook;
-#endif
-
-#else
 void __attribute__((constructor))
 __malloc_umem_init (void)
 {
 	umem_startup(NULL, 0, 0, NULL, NULL);
 }
-#endif
 
